packages/slate/src/changes/by-key.js

```diff
@@ -423,7 +423,7 @@ Changes.replaceNodeByKey = (change, key, newNode, options = {}) => {
   const parent = document.getParent(key)
   const index = parent.nodes.indexOf(node)
   change.removeNodeByKey(key, { normalize: false })
-  change.insertNodeByKey(parent.key, index, newNode, options)
+  change.insertNodeByKey(parent.key, index, newNode, { normalize: false })
   if (normalize) {
     change.normalizeNodeByKey(parent.key)
   }
```

Comment:
Yes, the effect would be the same but perhaps it would be a bit confusing. Each change has an "if normalize is true then normalize" so it might be confusing if this one doesn't have it. It also adds a level of indirection where you'd need to search for `insertNodeByKey`'s source code to make sure the normalization happens as expected.

Maybe I'm just overthinking this, what's your opinion?

Reply:
I would suggest `insertNodeByKey(parent.key, index, newNode, options)` or `insertNodeByKey(parent.key, index, newNode, {normalize})`, to be consistent with other places in code base